cmd/ddev/cmd/composer-createproject.go

```diff
@@ -0,0 +1,129 @@
+			}
+
+			positionals = append(positionals, arg)
+		}
+
+		// Package, directory, and version arguments are optional, positional, and difficult to parse.
+		// This will capture the possible argument combinations and assign them appropriately.
+		var pack, dir, ver string
+		if len(positionals) >= 3 {
+			pack = positionals[len(positionals)-3]
+			dir = positionals[len(positionals)-2]
+			ver = positionals[len(positionals)-1]
+		} else if len(positionals) == 2 {
+			pack = positionals[len(positionals)-2]
+			dir = positionals[len(positionals)-1]
+		} else if len(positionals) == 1 {
+			pack = positionals[len(positionals)-1]
+		}
+
+		app, err := ddevapp.GetActiveApp("")
+		if err != nil {
+			util.Failed(err.Error())
+		}
+
+		// Ensure project is running
+		if app.SiteStatus() != ddevapp.SiteRunning {
+			err = app.Start()
+			if err != nil {
+				util.Failed("Failed to start app %s to run create-project: %v", app.Name, err)
+			}
+		}
+
+		// Make the user confirm that existing contents will be deleted
+		// If the user hasn't passed any positional arguments, we're not actually executing an
+		// install and nothing will be deleted -- for example, 'ddev composer create-project -h'
+		doInstall := false
+		if len(positionals) > 0 {
+			util.Warning("Warning: Any existing contents of the project root will be overwritten")
+			doInstall = true
+			if !util.Confirm("Would you like to continue?") {
+				util.Failed("create-project cancelled")
+			}
+		}
+
+		// Only remove data if we expect an install
+		composerTmpDir := "/tmp/composer"
+		if doInstall {
+			// The install directory may be populated if a previous create-project was
+			// executed using the same container.
+			output.UserOut.Printf("Ensuring composer install directory is empty")
+			_, _, _ = app.Exec(&ddevapp.ExecOpts{
+				Service: "web",
+				Cmd:     []string{"sh", "-c", fmt.Sprintf("rm -rf %s", composerTmpDir)},
+			})
+
+			// Remove any contents of project root
+			util.Warning("Removing any existing files in project root")
+			_, _, _ = app.Exec(&ddevapp.ExecOpts{
+				Service: "web",
+				Cmd:     []string{"sh", "-c", "rm -rf /var/www/html/*"},
+			})
+		}
+
+		// Build container composer command
+		installDir := path.Join(composerTmpDir, dir)
+		composerCmd := append([]string{}, "composer", "create-project")
+		composerCmd = append(composerCmd, flags...)
+		composerCmd = append(composerCmd, pack, installDir, ver)
+
+		composerCmdString := strings.TrimSpace(strings.Join(composerCmd, " "))
+		output.UserOut.Printf("Executing composer command: %s\n", composerCmdString)
+		stdout, _, err := app.Exec(&ddevapp.ExecOpts{
+			Service: "web",
+			Cmd:     composerCmd,
+		})
+		if err != nil {
+			util.Failed("Failed to execute create-project command")
+		}
+
+		if len(stdout) > 0 {
+			fmt.Println(strings.TrimSpace(stdout))
+		}
+
+		// Move project to app root if an install is underway
+		if doInstall {
+			output.UserOut.Printf("Moving installation to project root")
```

Comment:
This one too might not be exactly correct. Not sure how we should work this in the very odd case where they don't do the project root. I'm kind of wondering if we should take away the option of *where* they can do the composer create-project, because if they don't use "." they're shooting themselves in the foot, and they're likely to use some example that messes them up. 

It's possible that we should adjust to the more normal examples, like `composer create-project typo3/cms-base-distribution CmsBaseDistribution ^9` and do all the things for them after that, like going into the dir, config. But that's too complex and too much change for where we are now. But people will definitely use examples like that one and get unexpected results that we'll have to support.